Tests/NIOTests/ChannelTests.swift

```diff
@@ -1716,4 +1716,92 @@ public class ChannelTests: XCTestCase {
             XCTFail("unexpected error: \(error)")
         }
     }
+
+    func testChannelReadsDoesNotHappenAfterRegistration() throws {
+        class ReadDoesNotHappen: ChannelInboundHandler {
+            typealias InboundIn = Any
+            private let hasRegisteredPromise: EventLoopPromise<()>
+            private let hasUnregisteredPromise: EventLoopPromise<()>
+            init(hasRegisteredPromise: EventLoopPromise<()>, hasUnregisteredPromise: EventLoopPromise<()>) {
+                self.hasRegisteredPromise = hasRegisteredPromise
+                self.hasUnregisteredPromise = hasUnregisteredPromise
+            }
+            func channelRead(ctx: ChannelHandlerContext, data: NIOAny) {
+                XCTFail("read has happened and shouldn't have")
+            }
+            func channelActive(ctx: ChannelHandlerContext) {
+                XCTFail("shouldn't have been activated")
+            }
+            func channelRegistered(ctx: ChannelHandlerContext) {
+                self.hasRegisteredPromise.succeed(result: ())
+            }
+            func channelUnregistered(ctx: ChannelHandlerContext) {
+                self.hasUnregisteredPromise.succeed(result: ())
+            }
+        }
+
+        let elg = MultiThreadedEventLoopGroup(numThreads: 1)
+        defer {
+            XCTAssertNoThrow(try elg.syncShutdownGracefully())
+        }
+        let sc = try SocketChannel(eventLoop: elg.next() as! SelectableEventLoop, protocolFamily: AF_INET)
+
+        class WriteImmediatelyHandler: ChannelInboundHandler {
+            typealias InboundIn = Any
+            typealias OutboundOut = ByteBuffer
+
+            private let writeDonePromise: EventLoopPromise<()>
+
+            init(writeDonePromise: EventLoopPromise<()>) {
```

Comment:
@weissi we usually use `EventLoopPromise<Void>` I think... Not sure what is the "more swifty way tho". Thoughts ?

Reply:
@normanmaurer for functions that have 'no return value' (sure you _could_ `return ()` but you don't have to) I tend to use `-> Void` but the promises we succeed with the value `()`. Therefore I mostly use `EventLoopPromise<()>` as there is a value 'sent'. Our codebase has both:

```
johannes:~/devel/swift-nio (jw-read-after-active-not-registered)
$ git grep 'EventLoopPromise<Void>' | wc -l
     220
johannes:~/devel/swift-nio (jw-read-after-active-not-registered)
$ git grep 'EventLoopPromise<()>' | wc -l
      57
```

Reply:
@weissi maybe a follow-up but I think w should make it consistent ;) I am not sure what is the "preferred" way tho.

Reply:
@normanmaurer ok, I'll file an issue.

Reply:
see #211